src/Diagnostics.DataProviders/Interfaces/IAppInsightsDataProvider.cs

```diff
@@ -8,5 +9,7 @@ public interface IAppInsightsDataProvider : IMetadataProvider
         Task<DataTable> ExecuteAppInsightsQuery(string query);
 
         Task<bool> SetAppInsightsKey(string appId, string apiKey);
+
+        Task<bool> InitializeAppInsightsKey(OperationContext<IResource> cxt);
```

Comment:
nitpick here :- Does it make sense to have both the method names same and use method overloading.

Reply:
Yup it does